src/global_scheduler/global_scheduler.c

```diff
@@ -86,6 +94,20 @@ void signal_handler(int signal) {
 
 /* End of the cleanup code. */
 
+local_scheduler *get_local_scheduler(global_scheduler_state *state,
+                                     db_client_id photon_id) {
+  local_scheduler *local_scheduler_ptr;
+  for (int i = 0; i < utarray_len(state->local_schedulers); ++i) {
+    local_scheduler_ptr =
+        (local_scheduler *) utarray_eltptr(state->local_schedulers, i);
+    if (memcmp(&local_scheduler_ptr->id, &photon_id, sizeof(photon_id)) == 0) {
```

Comment:
Can we use some kind of unique id comparison here? Maybe have a db_client_id comparison function.